gdx/src/com/badlogic/gdx/math/collision/OrientedBoundingBox.java

```diff
@@ -0,0 +1,142 @@
+	}
+
+	public Vector3 getCorner000 (final Vector3 out) {
+		return bounds.getCorner000(out).mul(transform);
+	}
+
+	public Vector3 getCorner001 (final Vector3 out) {
+		return bounds.getCorner001(out).mul(transform);
+	}
+
+	public Vector3 getCorner010 (final Vector3 out) {
+		return bounds.getCorner010(out).mul(transform);
+	}
+
+	public Vector3 getCorner011 (final Vector3 out) {
+		return bounds.getCorner011(out).mul(transform);
+	}
+
+	public Vector3 getCorner100 (final Vector3 out) {
+		return bounds.getCorner100(out).mul(transform);
+	}
+
+	public Vector3 getCorner101 (final Vector3 out) {
+		return bounds.getCorner101(out).mul(transform);
+	}
+
+	public Vector3 getCorner110 (final Vector3 out) {
+		return bounds.getCorner110(out).mul(transform);
+	}
+
+	public Vector3 getCorner111 (final Vector3 out) {
+		return bounds.getCorner111(out).mul(transform);
+	}
+
+	/** Returns whether the given vector is contained in this oriented bounding box.
+	 * @param v The vector
+	 * @return Whether the vector is contained or not. */
+	public boolean contains (Vector3 v) {
+		return contains(v, tmpMatrix.set(transform).inv());
+	}
+
+	private boolean contains (Vector3 v, Matrix4 invTransform) {
+		Vector3 localV = tmpVector.set(v).mul(invTransform);
+		return bounds.contains(localV);
+	}
+
+	/** Returns whether the given bounding box is contained in this oriented bounding box.
+	 * @param b The bounding box
+	 * @return Whether the given bounding box is contained */
+	public boolean contains (BoundingBox b) {
+		tmpMatrix.set(transform).inv();
+		return contains(b.getCorner000(tmpVector), tmpMatrix) && contains(b.getCorner001(tmpVector), tmpMatrix)
+			&& contains(b.getCorner010(tmpVector), tmpMatrix) && contains(b.getCorner011(tmpVector), tmpMatrix)
+			&& contains(b.getCorner100(tmpVector), tmpMatrix) && contains(b.getCorner101(tmpVector), tmpMatrix)
+			&& contains(b.getCorner110(tmpVector), tmpMatrix) && contains(b.getCorner111(tmpVector), tmpMatrix);
+	}
+
+	/** Returns whether the given oriented bounding box is contained in this oriented bounding box.
+	 * @param obb The oriented bounding box
+	 * @return Whether the given oriented bounding box is contained */
+	public boolean contains (OrientedBoundingBox obb) {
+		tmpMatrix.set(transform).inv();
+		return contains(obb.getCorner000(tmpVector), tmpMatrix) && contains(obb.getCorner001(tmpVector), tmpMatrix)
+			&& contains(obb.getCorner010(tmpVector), tmpMatrix) && contains(obb.getCorner011(tmpVector), tmpMatrix)
+			&& contains(obb.getCorner100(tmpVector), tmpMatrix) && contains(obb.getCorner101(tmpVector), tmpMatrix)
+			&& contains(obb.getCorner110(tmpVector), tmpMatrix) && contains(obb.getCorner111(tmpVector), tmpMatrix);
+	}
+
+	/** Returns whether the given bounding box is intersecting this oriented bounding box (at least one point in).
+	 * @param b The bounding box
+	 * @return Whether the given bounding box is intersected */
+	public boolean intersects (BoundingBox b) {
```

Comment:
Oops, my bad! I was doing it late night and trying to avoid SAT. Temporarily forgot about the edge cases.

I am thinking about creating an abstraction for SAT so it can be used for all kinds of intersections. 
Something like a 3D version of [this answer](https://jvm-gaming.org/t/having-problems-implementing-sat-separating-axis-theorem/58489/14).

```java
private static boolean separatingAxis(Vector2f[] v1s, Vector2f[] v2s, float dx, float dy) {
    float axisX = dy, axisY = -dx;
    float minA = Float.POSITIVE_INFINITY, maxA = Float.NEGATIVE_INFINITY;
    float minB = Float.POSITIVE_INFINITY, maxB = Float.NEGATIVE_INFINITY;
    int maxLen = Math.max(v1s.length, v2s.length);
    /* Project both polygons on axis */
    for (int k = 0; k < maxLen; k++) {
        if (k < v1s.length) {
            float d = v1s[k].x * axisX + v1s[k].y * axisY;
            if (d < minA) minA = d;
            if (d > maxA) maxA = d;
        }
        if (k < v2s.length) {
            float d = v2s[k].x * axisX + v2s[k].y * axisY;
            if (d < minB) minB = d;
            if (d > maxB) maxB = d;
        }
        /* Early-out if overlap found */
        if (minA <= maxB && minB <= maxA) {
            return false;
        }
    }
    return true;
}
public static boolean testPolygonPolygon(Vector2f[] v1s, Vector2f[] v2s) {
    int len1 = v1s.length, len2 = v2s.length;
    /* Try to find a separating axis using the first polygon's edges */
    for (int i = 0, j = len1 - 1; i < len1; j = i, i++)
        if (separatingAxis(v1s, v2s, v1s[i].x - v1s[j].x, v1s[i].y - v1s[j].y))
            return false; 
    /* Try to find a separating axis using the second polygon's edges */
    for (int i = 0, j = len2 - 1; i < len2; j = i, i++)
        if (separatingAxis(v1s, v2s, v2s[i].x - v2s[j].x, v2s[i].y - v2s[j].y))
            return false; 
    return true;
}
```